File: src/main/java/net/mreunionlabs/tapgradle/generate/generator/StructureGenerator.java
package net.mreunionlabs.tapgradle.generate.generator;

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension;
import net.mreunionlabs.tapgradle.generate.util.PackageDirectoryCreator;

import java.io.File;

/**
 * Created by abangkis on 3/19/2016.
 */
public class StructureGenerator {
    private static final String METAINF = "META-INF";

    public void createDirectories(GeneratePluginExtension ext) {
        File f = new File(ext.getJavaDir());
        f.mkdir();
        createResourceDir(ext);
        f = new File(ext.getWebAppDir());
        f.mkdir();
    }

    private void createResourceDir(GeneratePluginExtension ext) {
        File metaInf = new File(ext.getResDir(), METAINF);
        metaInf.mkdirs();

        File assets = new File(metaInf, "assets");
        assets.mkdir();

        // create css, images and js folder
        new File(assets, "css").mkdir();
        new File(assets, "images").mkdir();
        new File(assets, "js").mkdir();

        ext.setAssetFolder(assets);

        File modules = new File(metaInf, "modules");
        modules.mkdir();

        File resDir = new File(ext.getResDir());
        String packageString = ext.getPackageString();
        File packageRoot = PackageDirectoryCreator.createPackageDir(resDir, packageString);

        File components = new File(packageRoot, "components");
        components.mkdir();
        File pages = new File(packageRoot, "pages");
        pages.mkdir();

    }
}

File: src/main/kotlin/net/mreunionlabs/tapgradle/generate/generator/StructureGenerator.kt
package net.mreunionlabs.tapgradle.generate.generator

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension
import net.mreunionlabs.tapgradle.generate.util.PackageDirectoryCreator

import java.io.File

/**
 * Created by abangkis on 3/19/2016.
 */
class StructureGenerator {

    fun createDirectories(ext: GeneratePluginExtension) {
        var f = File(ext.javaDir)
        println("java dir ${f.absolutePath}")
        f.mkdir()
        createResourceDir(ext)
        f = File(ext.webAppDir)
        f.mkdir()
    }

    private fun createResourceDir(ext: GeneratePluginExtension) {
        val metaInf = File(ext.resDir, METAINF)
        metaInf.mkdirs()

        val assets = File(metaInf, "assets")
        assets.mkdir()

        // create css, images and js folder
        File(assets, "css").mkdir()
        File(assets, "images").mkdir()
        File(assets, "js").mkdir()

        ext.assetFolder = assets

        val modules = File(metaInf, "modules")
        modules.mkdir()

        val resDir = File(ext.resDir)
        val packageString = ext.packageString
        val packageRoot = PackageDirectoryCreator.createPackageDir(resDir, packageString)

        val components = File(packageRoot, "components")
        components.mkdir()
        val pages = File(packageRoot, "pages")
        pages.mkdir()

    }

    companion object {
        private val METAINF = "META-INF"
    }
}
